hotkey/windows.py

```diff
@@ -0,0 +1,359 @@
+        self.thread = None  # type: ignore
+        logger.debug('Done.')
+
+    def acquire_start(self) -> None:
+        """Start acquiring hotkey state via polling."""
+        pass
+
+    def acquire_stop(self) -> None:
+        """Stop acquiring hotkey state."""
+        pass
+
+    def fromevent(self, event) -> Optional[Union[bool, Tuple]]:  # noqa: CCR001
+        """
+        Return configuration (keycode, modifiers) or None=clear or False=retain previous.
+
+        event.state is a pain - it shows the state of the modifiers *before* a modifier key was pressed.
+        event.state *does* differentiate between left and right Ctrl and Alt and between Return and Enter
+        by putting KF_EXTENDED in bit 18, but RegisterHotKey doesn't differentiate.
+
+        :param event: tk event ?
+        :return: False to retain previous, None to not use, else (keycode, modifiers)
+        """
+        modifiers = ((GetKeyState(VK_MENU) & 0x8000) and MOD_ALT) \
+                    | ((GetKeyState(VK_CONTROL) & 0x8000) and MOD_CONTROL) \
+                    | ((GetKeyState(VK_SHIFT) & 0x8000) and MOD_SHIFT) \
+                    | ((GetKeyState(VK_LWIN) & 0x8000) and MOD_WIN) \
+                    | ((GetKeyState(VK_RWIN) & 0x8000) and MOD_WIN)
+        keycode = event.keycode
+
+        if keycode in [VK_SHIFT, VK_CONTROL, VK_MENU, VK_LWIN, VK_RWIN]:
+            return (0, modifiers)
+
+        if not modifiers:
+            if keycode == VK_ESCAPE:  # Esc = retain previous
+                return False
+
+            elif keycode in [VK_BACK, VK_DELETE, VK_CLEAR, VK_OEM_CLEAR]:  # BkSp, Del, Clear = clear hotkey
+                return None
+
+            elif keycode in [VK_RETURN, VK_SPACE, VK_OEM_MINUS] or ord('A') <= keycode <= ord(
+                'Z'):  # don't allow keys needed for typing in System Map
+                winsound.MessageBeep()
+                return None
+
+            elif (keycode in [VK_NUMLOCK, VK_SCROLL, VK_PROCESSKEY]
+                  or VK_CAPITAL <= keycode <= VK_MODECHANGE):  # ignore unmodified mode switch keys
+                return (0, modifiers)
+
+        # See if the keycode is usable and available
+        if RegisterHotKey(None, 2, modifiers | MOD_NOREPEAT, keycode):
+            UnregisterHotKey(None, 2)
+            return (keycode, modifiers)
+
+        else:
+            winsound.MessageBeep()
+            return None
+
+    def display(self, keycode, modifiers) -> str:
+        """
+        Return displayable form of given hotkey + modifiers.
+
+        :param keycode:
+        :param modifiers:
+        :return: string form
+        """
+        text = ''
+        if modifiers & MOD_WIN:
+            text += u'❖+'
+
+        if modifiers & MOD_CONTROL:
+            text += u'Ctrl+'
+
+        if modifiers & MOD_ALT:
+            text += u'Alt+'
+
+        if modifiers & MOD_SHIFT:
+            text += u'⇧+'
+
+        if VK_NUMPAD0 <= keycode <= VK_DIVIDE:
+            text += u'№'
```

Comment:
```suggestion
        if modifiers & MOD_WIN:
            text += '❖+'

        if modifiers & MOD_CONTROL:
            text += 'Ctrl+'

        if modifiers & MOD_ALT:
            text += 'Alt+'

        if modifiers & MOD_SHIFT:
            text += '⇧+'

        if VK_NUMPAD0 <= keycode <= VK_DIVIDE:
            text += '№'
```

Unicode strings do nothing in py3

Reply:
Legacy code ahoy !   Fixed in this file.  I'm going to hope that *when* I get around to trying `black` again it would take care of this sort of thing ?

Reply:
```
--- test.py     2022-12-23 12:26:01.724569 +0000
+++ test.py     2022-12-23 12:26:07.943035 +0000
@@ -1 +1 @@
-u"test"
+"test"
would reformat test.py

All done! ✨ 🍰 ✨
1 file would be reformatted.
```
yep!

Reply:
OK, I do intend another stab at black, and trying really hard to not get angry with it messing anything up ;).